app/models/company.rb

```diff
@@ -37,13 +37,31 @@ class Company < ApplicationRecord
   has_many :pictures, class_name: 'Ckeditor::Picture', dependent: :destroy
 
   accepts_nested_attributes_for :addresses, allow_destroy: true
+  alias_method :categories, :business_categories
+  delegate :visible, to: :addresses, prefix: true
 
   def approved_applications_from_members
     # Returns ActiveRecord Relation
     shf_applications.accepted.includes(:user)
       .order('users.last_name').where('users.member = ?', true)
   end
 
+  def any_visible_addresses?
+    addresses_visible.any?
+  end
+
+  def categories_names
+    categories.select(:name).distinct.order(:name).pluck(:name)
+  end
+
+  def addresses_region_names
+    addresses.joins(:region).select('regions.name').distinct.pluck('regions.name')
+  end
+
+  def cities_names
```

Comment:
Cities and kommuns are not the same.  Check with @thesuss.  These are typically referred to as "municipalities" in English.  However, since that is an awkward word to use we have just been using "kommun" (or its plural). 

So, perhaps this method should be named `kommuns_names`.